src/index.ts

```diff
@@ -109,10 +111,57 @@ export class EppoJSClient extends EppoClient {
     isObfuscated: true,
   });
 
+  /**
+   * @deprecated
+   */
   public static initialized = false;
 
   private initialized = false;
 
+  public static buildAndInit(config: IClientConfig): EppoJSClient {
+    const flagConfigurationStore =
+      config.flagConfigurationStore ??
+      configurationStorageFactory({
+        forceMemoryOnly: true,
+      });
+    const client = new EppoJSClient({ flagConfigurationStore });
+
+    // init will resolve the promise that client.waitForConfiguration returns.
+    client.init(config);
+    return client;
+  }
+
+  /**
+   * Resolved when the client is initialized
+   * @private
+   */
+  private readonly initializedPromise: Promise<void>;
```

Comment:
nit: i'd rename to `initializationPromise` or simply `initPromise`

Reply:
good point. It's not an initialized promise, it's a promise signalling when it is initialized.